jdaviz/configs/default/plugins/plot_options/plot_options.py

```diff
@@ -1087,3 +1111,13 @@ def _is_image_viewer(viewer):
             viewers = [viewers]
 
         return np.all([_is_image_viewer(viewer) for viewer in viewers])
+
+    @observe('image_colormap_value')
+    def _random_cmap_limit_update(self, *args, **kwargs):
+        # if 'Random' colormap is used for visualizing image segmentation,
+        # ensure the stretch limits are the min and max, so all label colors
+        # are unique:
+        if self.image_colormap_value != 'Random':
+            return
+
+        self.stretch_preset.value = 100

```

Comment:
I think I would prefer this to just toggle an in-plugin warning.  I can even see how it would be useful to set non-min/max in order to "exclude" outliers, but I can see how that at least needs some sort of explanatory message and perhaps link to a section in the docs explaining this implementation and the nuances.

Reply:
@kecnry For a segmentation image, the default here needs to be min/max.  Nothing else really makes sense.  Having a non-min/max would clip a lower and upper range of integer labels (which are generally at the bottom and top of the segmentation image) and thus different sources would get assigned the same colors.  There is no "outlier values" in a segmentation image.  Each source gets assigned a unique integer value. 

As an example of why non-min/max is very bad, the background (no detected sources) is assigned a label value of zero. If you were to assign say all values < 10 to the same color, then you couldn't even distinguish between background and sources, defeating the whole point.

This must use min/max.  An appropriate message can be shown stating the stretch levels have been changed to min/max.

Reply:
Ok, then can we detect that the image is a segmentation image and only show this as an option in that case?  I'm just worried about the more common case where an image is _not_ a segmentation image, the user selects "random" not knowing the implications, and things are changed automatically and perhaps they loose their carefully set stretch limits.

Reply:
Segmentation images contain only integer data, but that's not a unique identifier.  Not sure of any way to automatically detect a segmentation image.

Reply:
Segmentation images often contain consecutive integer values (0 to max label), but that's definitely not a requirement either.

Reply:
Maybe this would be better as a separate plugin (or button within plot options that explicitly states all the things that will be set - similar to the RGB defaults button) then?  Otherwise, I think we're adding confusion for the default case in order to support a case that isn't relevant for most situations.

Reply:
> One plugin which controls plot options is Plot Options.

I know, and I agree it makes most sense here, but then it needs to act in a way that makes sense for any input data in which the choice is available.  If you're someone who isn't even familiar with segmentation maps and load an image and stumble on this, it will be very jarring and confusing.

> If you're intentionally using this button for its intended purpose, this is adding clicks that you will always have to go through.

Yes, that is the downside, but I think it makes sense to add the extra clicks to the specific use-case rather than the general use-case.  That said, the default stretch function is already linear, so by default this will just be two clicks, not three.  If the parser could detect segmentation maps, then we could even have the percentile default for those layers to min/max from load, and then this only requires extra clicks if the user overrides anything.

Another benefit to this design is if the user changes to "Image Segmentation" (or "Random") and _then_ changes the stretch inputs, the warning will re-appear.  The current implementation has no way to catch that case unless it disables those inputs entirely.

> Would your concerns be assuaged if we named the cmap "Image Segmentation"?

It is slightly more specific than "Random", but I still don't think it provides the necessary context and explanation for someone who does not have this use-case in their mind at all when looking through the list of options for colormap.